packages/@uppy/companion/src/server/provider/credentials.js

```diff
@@ -137,8 +138,10 @@ exports.getCredentialsOverrideMiddleware = (providers, companionOptions) => {
         },
       }
 
-      if (credentials.redirect_uri) {
-        res.locals.grant.dynamic.redirect_uri = credentials.redirect_uri
+      if (credentials.transloadit_gateway) {
+        const redirectPath = getRedirectPath(providerName)
+        const fullRedirectPath = getURLBuilder(companionOptions)(redirectPath, true)
+        res.locals.grant.dynamic.redirect_uri = `${credentials.transloadit_gateway}${fullRedirectPath}`
```

Comment:
If you want to prevent a double `/` when `transloadit_gateway` already contains one
```suggestion
        res.locals.grant.dynamic.redirect_uri = new URL(fullRedirectPath, credentials.transloadit_gateway).toString()
```